### ISSUE TITLE
DXF spline import tied to frame settings. Cannot be used headless.

### ISSUE THREAD

**Issue by ChrisClems** [FriendsOfCADability/CADability#187]
On line 533-534, ImportDxf is attempting to get the precision value to approximate a spline: https://github.com/FriendsOfCADability/CADability/blob/ddb68c06672904949ac7961aba92c6ddff4078fa/CADability/ImportDxf.cs#L533

It is using Project.Frame.GetDoubleSetting(). In a headless environment, Project.Frame will be null so it is throwing an exception and failing to import the DXF when a spline that triggers this line of code is encountered. It works fine when loading the DXF through CADAbility.App.

I removed the rest of the original comment on troubleshooting this issue as it appears to be a bug in Rider. Program works as expected in Visual Studio when hard-coding the maxError value.

**Comment by dsn27** [comment]
GetDoubleSetting is implemented like this.

```
public double GetDoubleSetting(string Name, double Default)
{
    object s = GetSetting(Name);
    if (s != null)
    {
        try
        {
            return Settings.GetDoubleValue(s);
        }
        catch (InvalidCastException)
        {
            return Default;
        }
    }
    return Default;
}
```

You could create a new function in ImportDxf to help you replace project.Frame.GetDoubleSetting(...).
If this works fine, a PR would be appreciated to fix this use case.

Thanks in advance,
Michel

**Comment by ChrisClems** [comment]
> GetDoubleSetting is implemented like this.
> 
> ```
> public double GetDoubleSetting(string Name, double Default)
> {
>     object s = GetSetting(Name);
>     if (s != null)
>     {
>         try
>         {
>             return Settings.GetDoubleValue(s);
>         }
>         catch (InvalidCastException)
>         {
>             return Default;
>         }
>     }
>     return Default;
> }
> ```
> 
> You could create a new function in ImportDxf to help you replace project.Frame.GetDoubleSetting(...). If this works fine, a PR would be appreciated to fix this use case.
> 
> Thanks in advance, Michel

I'll play around with in over the next couple of weeks and find a comfortable pattern. I'm doing a lot of work with DXF import and export for a project and I'm hitting a lot of little limitations that I plan to submit PRs for as I have time to test them.

**Comment by ChrisClems** [comment]
Turned out to only take a couple of hours. I used the GlobalSettings class to re-use existing functionality without having to change the signature for the DXF export or make it difficult to set on the user side.

In the process I found a bug in SetValue() that wasn't handling DoubleProperties. I added a handler so all DoubleProperties in the GlobalSettings can now be set via code. There might be more missing.

When digging into the GlobalSettings structure, I found more references to Project.Frame for properties. Might be worth opening a new issue just to find them and convert them to library-accessible formats.
